internal/bminventory/inventory_v2_handlers.go

```diff
@@ -671,6 +671,29 @@ func (b *bareMetalInventory) V2SetIgnoredValidations(ctx context.Context, params
 	return installer.NewV2SetIgnoredValidationsCreated().WithPayload(&ignoredValidations)
 }
 
+func (b *bareMetalInventory) V2GetClusterInstallerUISettings(ctx context.Context, params installer.V2GetClusterInstallerUISettingsParams) middleware.Responder {
+	cluster, err := b.getCluster(ctx, params.ClusterID.String())
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return installer.NewV2GetClusterInstallerUISettingsNotFound().WithPayload(common.GenerateError(http.StatusNotFound, err))
+		}
+		err = errors.Wrapf(err, "failed to fetch cluster %s to get UI settings", *cluster.ID)
+		return installer.NewV2GetClusterInstallerUISettingsInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
```

Comment:
Please use [`common.GenerateErrorResponder`](https://github.com/openshift/assisted-service/blob/3756e7f3c06bdc66446f29bb9f600af0d8bd6fb2/internal/common/error_utils.go#L118-L133) this handles returning 404 for gorm not found errors as well as some other cases with wrapped errors.